From 5d2216514461f92fbaed1106546081b518362f9c Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Fri, 13 Feb 2015 11:52:57 +0530
Subject: [PATCH 008/132] MIPS/EDAC: Poll for LMC_INT_REG[nxm_wr_err]

Commit 6cd6d5cba789f5e3afef21ec53a2873027e14ba3 from
git://git.yoctoproject.org/linux-yocto-3.14

Add missing error interrupt, nxm_wr_err, write to no-existent memory.

MIPS/EDAC: Set error reporting state to polling

For LMC controller set error reporting state to EDAC_OPSTATE_POLL.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Signed-off-by: Abhishek Paliwal <abhishek.paliwal@aricent.com>
[Original patch taken from OCTEON-SDK 3.1.1-544.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/edac/octeon_edac-lmc.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/edac/octeon_edac-lmc.c b/drivers/edac/octeon_edac-lmc.c
index 821462e..c9b2f9e 100644
--- a/drivers/edac/octeon_edac-lmc.c
+++ b/drivers/edac/octeon_edac-lmc.c
@@ -116,6 +116,14 @@ static void octeon_lmc_edac_poll_o2(struct mem_ctl_info *mci)
 		do_clear = true;
 	}
 
+	if (int_reg.s.nxm_wr_err) {
+		snprintf(msg, sizeof(msg), "NXM_WR_ERR: Write to non-existent memory");
+		edac_mc_handle_error(HW_EVENT_ERR_CORRECTED, mci, 1, 0, 0, 0,
+					-1, -1, -1, msg, "");
+		int_reg.s.nxm_wr_err = -1;      /* Done, re-arm */
+		do_clear = true;
+	}
+
 	if (do_clear) {
 		if (likely(!pvt->inject))
 			cvmx_write_csr(CVMX_LMCX_INT(mci->mc_idx), int_reg.u64);
@@ -234,6 +242,8 @@ static int octeon_lmc_edac_probe(struct platform_device *pdev)
 	layers[0].size = 1;
 	layers[0].is_virt_csrow = false;
 
+	edac_op_state = EDAC_OPSTATE_POLL;
+
 	if (OCTEON_IS_OCTEON1PLUS()) {
 		union cvmx_lmcx_mem_cfg0 cfg0;
 
-- 
1.9.1

